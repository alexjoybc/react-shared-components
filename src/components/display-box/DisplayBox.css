.display-box {
  display: flex;
  flex-direction: row;
  padding: 15px;
}

.no-padding-bottom {
  padding: 10px;
  padding-bottom: 0px !important;
}

.display-icon {
  padding-right: 15px;
}

.display-right-element {
  width: 100%;
  padding-top: 3px;
}

.display-left-element {
  padding-left: 0px !important;
}

.grey-background {
  background-color: lightgrey;
}

.blue-background {
  background-color: rgba(225, 232, 242, 1);
}

.border-background {
  border-width: thin;
  border-style: solid;
  border-color: lightgrey;
}

.react-icons {
  vertical-align: middle;
}
