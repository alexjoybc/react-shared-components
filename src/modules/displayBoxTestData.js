import React from "react";
import { Table } from "../components/table/Table";
import { getTableElementsTestData } from "./tableTestData";

export function getTestTable() {
  const tableTestData = getTableElementsTestData();
  return <Table elements={tableTestData} />;
}
