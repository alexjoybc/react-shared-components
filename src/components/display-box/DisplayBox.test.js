import React from "react";
import { MdPerson } from "react-icons/md";

import testBasicSnapshot from "../../TestHelper";
import { DisplayBox } from "./DisplayBox";
import { getTestTable } from "../../modules/displayBoxTestData";

const table = getTestTable();
const icon = <MdPerson size={32} />;

describe("DisplayBox Component", () => {
  test("Matches the 'without icon' snapshot", () => {
    testBasicSnapshot(<DisplayBox element={table} />);
  });

  test("Matches the 'with icon' snapshot", () => {
    testBasicSnapshot(<DisplayBox icon={icon} element={table} />);
  });

  test("Matches the warning snapshot", () => {
    testBasicSnapshot(
      <DisplayBox styling={"warning-background"} icon={icon} element={table} />
    );
  });

  test("Matches the blue-background snapshot", () => {
    testBasicSnapshot(
      <DisplayBox styling={"blue-background"} icon={icon} element={table} />
    );
  });
});
