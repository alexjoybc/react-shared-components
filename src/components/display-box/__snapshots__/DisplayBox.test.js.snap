// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayBox Component Matches the 'with icon' snapshot 1`] = `
<div
  className="display-box "
>
  <div
    className="display-icon"
  >
    <svg
      fill="currentColor"
      height={32}
      size={32}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={32}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
  <div
    className="display-right-element"
  >
    <table
      className=""
    >
      <thead>
        <tr>
          <th
            colSpan="2"
          >
            
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className=""
          colSpan="2"
        >
          <td
            className=""
          >
            BCeID
          </td>
          <td>
            bobross42
          </td>
        </tr>
        <tr
          className=""
          colSpan="2"
        >
          <td
            className=""
          >
            Last Name
          </td>
          <td>
            Ross
          </td>
        </tr>
        <tr
          className=""
          colSpan="2"
        >
          <td
            className=""
          >
            First Name
          </td>
          <td>
            Bob
          </td>
        </tr>
        <tr
          className=""
          colSpan="2"
        >
          <td
            className=""
          >
            Email Address
          </td>
          <td>
            bob.ross@example.com
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`DisplayBox Component Matches the 'without icon' snapshot 1`] = `
<div
  className="display-box "
>
  <div
    className="display-right-element"
  >
    <table
      className=""
    >
      <thead>
        <tr>
          <th
            colSpan="2"
          >
            
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className=""
          colSpan="2"
        >
          <td
            className=""
          >
            BCeID
          </td>
          <td>
            bobross42
          </td>
        </tr>
        <tr
          className=""
          colSpan="2"
        >
          <td
            className=""
          >
            Last Name
          </td>
          <td>
            Ross
          </td>
        </tr>
        <tr
          className=""
          colSpan="2"
        >
          <td
            className=""
          >
            First Name
          </td>
          <td>
            Bob
          </td>
        </tr>
        <tr
          className=""
          colSpan="2"
        >
          <td
            className=""
          >
            Email Address
          </td>
          <td>
            bob.ross@example.com
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`DisplayBox Component Matches the blue-background snapshot 1`] = `
<div
  className="display-box blue-background"
>
  <div
    className="display-icon"
  >
    <svg
      fill="currentColor"
      height={32}
      size={32}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={32}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
  <div
    className="display-right-element"
  >
    <table
      className=""
    >
      <thead>
        <tr>
          <th
            colSpan="2"
          >
            
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className=""
          colSpan="2"
        >
          <td
            className=""
          >
            BCeID
          </td>
          <td>
            bobross42
          </td>
        </tr>
        <tr
          className=""
          colSpan="2"
        >
          <td
            className=""
          >
            Last Name
          </td>
          <td>
            Ross
          </td>
        </tr>
        <tr
          className=""
          colSpan="2"
        >
          <td
            className=""
          >
            First Name
          </td>
          <td>
            Bob
          </td>
        </tr>
        <tr
          className=""
          colSpan="2"
        >
          <td
            className=""
          >
            Email Address
          </td>
          <td>
            bob.ross@example.com
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`DisplayBox Component Matches the warning snapshot 1`] = `
<div
  className="display-box warning-background"
>
  <div
    className="display-icon"
  >
    <svg
      fill="currentColor"
      height={32}
      size={32}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={32}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
  <div
    className="display-right-element"
  >
    <table
      className=""
    >
      <thead>
        <tr>
          <th
            colSpan="2"
          >
            
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className=""
          colSpan="2"
        >
          <td
            className=""
          >
            BCeID
          </td>
          <td>
            bobross42
          </td>
        </tr>
        <tr
          className=""
          colSpan="2"
        >
          <td
            className=""
          >
            Last Name
          </td>
          <td>
            Ross
          </td>
        </tr>
        <tr
          className=""
          colSpan="2"
        >
          <td
            className=""
          >
            First Name
          </td>
          <td>
            Bob
          </td>
        </tr>
        <tr
          className=""
          colSpan="2"
        >
          <td
            className=""
          >
            Email Address
          </td>
          <td>
            bob.ross@example.com
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
